chainer/initializations.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import absolute_import
+
+import numpy as np
+
+# Original code forked from MIT licensed keras project
+# https://github.com/fchollet/keras/blob/master/keras/initializations.py
+
+
+def init_weight(weights, initWith, scale=1,
+                none_default=lambda shape: he_normal(shape, 1 / np.sqrt(2))):
+    """Initializes the given weight matrix.
+
+    Args:
+            weights (~numpy.ndarray): Weight matrix to be initialized.
+            initWith (value): The value to use to initialize the data. May be
+                `None` (in which case the none_default value will be used), a
+                scalar to set all values to, a matrix of the same shape to
+                copy, or a callable that takes the shape of the matrix as input
+                and returns a matrix of the correct size.
+            scale (scalar): A constant to multiply initWith by, if initWith was
+                a callable.
+            none_default (value): Same as initWith, the value here must not be
+                `None`, as it will be used as a default when initWith is
+                `None`.
+
+    """
+    if initWith is None:
+        initWith = none_default
+    if callable(initWith):
+        initWith = scale * initWith(weights.shape)
+    if hasattr(initWith, 'shape'):
+        # check needed for bilinear tests to pass
+        assert weights.shape == initWith.shape
```

Comment:
Personally, I'm not positive on where it would be useful to have an initialization that is broadcastable with the correct weight. If a matrix is being passed in, the only logical assumption that makes sense to me (IMHO) is that the caller wants this specific weight matrix to be the starting value of the weights. 

So if we are loading saved weights from another run, it should never be an issue. If for some reason they are manually constructing a matrix to use, I see no reason that the user shouldn't make their code produce a matrix of the correct shape. My personal intuition is that if you allow it to be broadcastable too, it may occasionally result in unintuitive behavior where something initializes that one would more logically expect to error out.